### ISSUE TITLE
difference between using train from scratch and finetuning 

### ISSUE THREAD

**Issue by alaa-shubbak** [open-mmlab/mmdetection#5595]
Hello , 

I would like to ask regard the difference (in coding) , between do training from scratch and make finetuning and transfer learning. 

I read through the mmdetection document , but it dose not mentioned . 

I understand it like this was: 
training from scratch (beginning ) use this instruction 

`python tools/train.py configs/ssd/ssd512_coco_my.py --gpus 2 --work-dir training_data `

while do the finetuning  used the following; 
`python tools/train.py configs/ssd/ssd512_coco_my.py --gpus 2 --work-dir training_data --resume-from checkpoints/ssd512_coco_20200308-038c5591.pth
`

is that correct? 

I did the training using similar code (training from scratch ) it gives me this: 

![traning](https://user-images.githubusercontent.com/10245810/125320929-78d99380-e33c-11eb-8280-af4f82e6dac9.JPG)

while in finetuning as i understood , it gives me this 
![after training ssd](https://user-images.githubusercontent.com/10245810/125320986-8a22a000-e33c-11eb-9997-96dcf9f6b18f.JPG)

is that all correct? is my concept of fine tuning and training correct? 

in the 2nd training , i got only two files , one a (.json) file and the other (.log), is it normal? and why not getting (.pth) which i can use it later to test my model on the validation dataset? 

looking forward to hear from any of you.

**Comment by noobying** [comment]
In the 2nd training, I think you should run with --load-from(just load the model weight) instead of --resume-from(load extra training information,e.g. resume epoch, learning rate..)

**Comment by alaa-shubbak** [comment]
> In the 2nd training, I think you should run with --load-from(just load the model weight) instead of --resume-from(load extra training information,e.g. resume epoch, learning rate..)

--load-from as an argument is not acceptable. have a look 
![error load from](https://user-images.githubusercontent.com/10245810/125395757-e244bb00-e3ab-11eb-990d-347f356a2e79.JPG)

also , it can not work alone without any train or test function.

**Comment by alaa-shubbak** [comment]
i used --load-from in the script of my configuration as bellow; 

![Uploading Capture.JPG…]()

and i think this is the only way to use it.

**Comment by alaa-shubbak** [comment]
Thank you , it is working correctly now with me. 

I did the training on CenterNet model , with backbone ResNet 18. 

here is the training command 
`python tools/train.py configs/human/centernet_resnet18_140e_coco.py --gpus 1 --work-dir training_data/centernet_epoch12`

i did it with 12 epoch  and 1 iteration. 

I got a 12 zip file with extension of (.pth) , it is strange for me to get them in a form of zip file? 
have a look on what produced in my work-dir
![content of my dir after training](https://user-images.githubusercontent.com/10245810/125649784-f2f464c1-7476-4203-83f3-062aab78d1d3.JPG)


when I try to test the new model with such new weight in .pth file , it gives me error. have a look 

![error in testing](https://user-images.githubusercontent.com/10245810/125649876-194cdc34-66a0-418e-8a01-da39557cd7c5.JPG)

what do you think? any help please

**Comment by noobying** [comment]
It‘s okay to see pth file with zip icon. Have a look for my work dir.
![image](https://user-images.githubusercontent.com/42159793/125702681-f1c5f11b-e662-4701-8cb9-b7ae22963fb7.png)
It seems you lost the ori_filename in the img_meta. By the way, file_name is a key assigned in the 'LoadImageFromFile'.
https://github.com/open-mmlab/mmdetection/blob/5f6134796ed07f443d1bf389f57d3cee40bbd044/mmdet/datasets/pipelines/loading.py#L65
You can debug whether you modify  the 'ori_filename' in the test phase.

**Comment by alaa-shubbak** [comment]
> It‘s okay to see pth file with zip icon. Have a look for my work dir.
> ![image](https://user-images.githubusercontent.com/42159793/125702681-f1c5f11b-e662-4701-8cb9-b7ae22963fb7.png)
> It seems you lost the ori_filename in the img_meta. By the way, file_name is a key assigned in the 'LoadImageFromFile'.
> https://github.com/open-mmlab/mmdetection/blob/5f6134796ed07f443d1bf389f57d3cee40bbd044/mmdet/datasets/pipelines/loading.py#L65
> 
> 
> You can debug whether you modify the 'ori_filename' in the test phase.


First of all , Thank you so much for your replay. it makes me clam as i was thinking that i am doing something wrong. I was almost to quit working with mmedetion cause of difficulties in implementations 

I still have two questions, 
1. where shall i change/modify  'ori_filename' and this meta data ? 
I used coco dataset  type , shall i change it in the dataset (mmdetection/configs/_base_/datasets/coco_detection.py ) or shall i change it  in the produced python file (in my images it is called "centernet_resnet18_140e_coco.py") or in my configuration? 

here is a place where I can find more info about meta-data , it is in  my configuration file only. 

![meta_data](https://user-images.githubusercontent.com/10245810/125748092-c1164f26-5852-4b75-a399-68bc6161a926.JPG)

shall I add 'ori_filename' in this structure array of  meta-key? 

2. Is there any other way of testing such results (.pth) better than my way? 

many thanks.

**Comment by noobying** [comment]
Have you ever modified the code in your project?

**Comment by alaa-shubbak** [comment]
> Have you ever modified the code in your project?

yes , but not every thing in the code ,  I only modified the part related to the classes in my data. instead of 80 classes in coco dataset , i only used one classes. 
other things i kept them as it is.

**Comment by noobying** [comment]
Did you only modify the config file? Could you please show the full config?

**Comment by alaa-shubbak** [comment]
> Did you only modify the config file? Could you please show the full config?

yes, I updated  the classes to become classes=('Person',)  as I want to train CenterNet on my new dataset (which is in a form  similar to coco dataset)
I also change the num_classes in the config to be 1 

here is the full config , it has dataset , pipeline for train , pipeline for test, train dir ,test dir , val dir , and evaluation ,checkpoint details. 

<img width="455" alt="config1" src="https://user-images.githubusercontent.com/10245810/125823617-763ed989-3d3a-433b-bbe1-9e0457ea2296.png">

![test pipeline](https://user-images.githubusercontent.com/10245810/125823947-7cb15059-28cb-4a0f-b6b1-89d9239f796f.JPG)

![datadic](https://user-images.githubusercontent.com/10245810/125823975-68e63878-a686-4fb1-9bf9-967d567f0f24.JPG)

![valdic](https://user-images.githubusercontent.com/10245810/125824053-76fca48d-c8b2-4492-9a6f-9d27b1b9663c.JPG)

![testdic](https://user-images.githubusercontent.com/10245810/125824064-bd5bdf3d-7c96-45c6-8afc-ae8de83d407f.JPG)

![evaluation](https://user-images.githubusercontent.com/10245810/125824079-4aff5d55-ff69-450a-9338-1c9a6752e9b8.JPG)

**Comment by alaa-shubbak** [comment]
i got it , thank you so much
